src/app/shared-services/trajectory-import-export.service.ts

```diff
@@ -0,0 +1,281 @@
+const { FileSelector, Filesystem } = Plugins
+
+@Injectable()
+export class TrajectoryImportExportService extends TrajectoryService {
+  constructor(
+    http: HttpClient,
+    db: SqliteService,
+    locationService: LocationService,
+    private socialSharing: SocialSharing,
+    private loadingController: LoadingController,
+    private toastController: ToastController,
+    private actionSheetController: ActionSheetController,
+    private androidPermissions: AndroidPermissions,
+    private platform: Platform
+  ) {
+    super(http, db, locationService)
+  }
+
+  /**
+   * Invokes a UI that enables the user to select and import
+   * a trajectory-json on iOS and Android.
+   */
+  async selectAndImportTrajectory() {
+    const selectedFile = await FileSelector.fileSelector({
+      multiple_selection: false,
+      ext: ['*'],
+    })
+    if (this.platform.is('android')) {
+      const parsedPaths = JSON.parse(selectedFile.paths)
+      const parsedOriginalNames = JSON.parse(selectedFile.original_names)
+      const parsedExtensions = JSON.parse(selectedFile.extensions)
+      for (let index = 0; index < parsedPaths.length; index++) {
+        const file = await fetch(parsedPaths[index]).then((r) => r.blob())
+        await this.importFile(
+          file,
+          parsedOriginalNames[index],
+          parsedExtensions[index]
+        )
+      }
+    } else if (this.platform.is('ios')) {
+      for (let index = 0; index < selectedFile.paths.length; index++) {
+        const file = await fetch(selectedFile.paths[index]).then((r) =>
+          r.blob()
+        )
+        await this.importFile(
+          file,
+          selectedFile.original_names[index],
+          selectedFile.extensions[index]
+        )
+      }
+    }
+  }
+
+  /**
+   * Translates a trajectory in form of a JSON {@linkcode TrajectoryJSON}
+   * into a model class {@linkcode Trajectory}
+   *
+   * @param json trajectory-file as JSON, see {@linkcode TrajectoryJSON}
+   * @param name file name of imported JSON
+   *
+   * @returns trajectory
+   */
+  createTrajectoryFromImport(json: string, name: string): Trajectory {
+    const trajectoryJson: {
+      coordinates: string
+      timestamps: number[]
+      time0: string
+      timeN?: string
+    } = JSON.parse(json)
+    const data = Trajectory.fromJSON(trajectoryJson)
+    const meta: TrajectoryMeta = {
+      id: uuid(),
+      placename: name?.replace(/\.[^/.]+$/, '') ?? 'trajectory', // remove extension from name (e.g. '.json')
+      type: TrajectoryType.IMPORT,
+      durationDays: null,
+    }
+    return new Trajectory(meta, data)
+  }
+
+  /**
+   * Loads full trajectory from persistent storage and invokes a UI
+   * that enables the user to export a trajectory as JSON.
+   *
+   * @param trajectoryMeta metadata of a trajectory
+   */
+  async exportTrajectory(trajectoryMeta: TrajectoryMeta) {
+    this.getOne(trajectoryMeta.type, trajectoryMeta.id).subscribe(async (t) => {
```

Comment:
We don't want to `.subscribe()` here.
```suggestion
    this.getOne(trajectoryMeta.type, trajectoryMeta.id).subscribe(async (t) => {
    const t = await this.getOne(trajectoryMeta.type, trajectoryMeta.id).toPromise()
```
Also, this operation may take a while (try exporting imported nepal example), a loading spinner should be displayed